### ISSUE TITLE
Find most expensive hours

### ISSUE THREAD

**Issue by Gr1pen** [ottopaulsen/node-red-contrib-power-saver#159]
I would really like an option to find the most expensive hours.
The use case is charging a home battery the cheapest hours, and use that energy the most expensive hours.

**Comment by shadowano** [comment]
If you are already finding cheapest hours, would the non-cheapest hours be good to say that they are the most expensive hours?

This way you could treat the off signal as a start of expensive hours and on signal as stop for expensive hours. Use a function node to for instance invert the signals. 

You could possibly also use a lowest price node and adjust the min hours to be on to a high enough value so that only the most expensive hours are left, then invert the signal on the node. 

This would of course not work optimally if the price is flat during a day. In best price node you can set which price you care about which could solve this

**Comment by Gr1pen** [comment]
Yes I could probably create a kind of inverted output. Thanks.

 

Still it would be nice to be able to find the most expensive hours the same
way it works with cheapest hours today.

 

                                                                           

Med vänlig hälsning,
Tonny Andersson


0766-225 123 | ***@***.***
***@***.***> 
Högs Mellanväg 103-22 | 244 93 Kävlinge

 

Från: shadowano ***@***.***> 
Skickat: den 13 mars 2023 08:00
Till: ottopaulsen/node-red-contrib-power-saver
***@***.***>
Kopia: Tonny Andersson ***@***.***>; Author
***@***.***>
Ämne: Re: [ottopaulsen/node-red-contrib-power-saver] Find most expensive
hours (Issue #159)

 

If you are already finding cheapest hours, would the non-cheapest hours be
good to say that they are the most expensive hours?

This way you could treat the off signal as a start of expensive hours and on
signal as stop for expensive hours. Use a function node to for instance
invert the signals.

You could possibly also use a lowest price node and adjust the min hours to
be on to a high enough value so that only the most expensive hours are left,
then invert the signal on the node.

This would of course not work optimally if the price is flat during a day.
In best price node you can set which price you care about which could solve
this


Reply to this email directly, view it on GitHub
<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.co
m%2Fottopaulsen%2Fnode-red-contrib-power-saver%2Fissues%2F159%23issuecomment
-1465607663&data=05%7C01%7Ctonny.andersson%40thunder-cloud.com%7C5473c4c0e03
b4f77918508db23909da0%7Ce56f597361424d3493f629f8eece565e%7C0%7C0%7C638142876
259836261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTi
I6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=c9EU1jfFPYRq2yO1jn6ZbbJbcjOX7
V9qxphSlmuzOK0%3D&reserved=0> , or unsubscribe
<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.co
m%2Fnotifications%2Funsubscribe-auth%2FAKAP3LEOU7LAOM23WL5JDQTW33AYLANCNFSM6
AAAAAAVPNT234&data=05%7C01%7Ctonny.andersson%40thunder-cloud.com%7C5473c4c0e
03b4f77918508db23909da0%7Ce56f597361424d3493f629f8eece565e%7C0%7C0%7C6381428
76259836261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJB
TiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=OSrStiFVhdt3%2BWalMyTNuEXvD
C4sdwrKvCj6gMGNGbc%3D&reserved=0> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/AKAP3LFXOFVWJWB3V67ZULDW33AYLA5CNFS
M6AAAAAAVPNT236WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTS
XLNS66.gif> Message ID:
***@***.***
***@***.***
b.com> >

**Comment by gittan443** [comment]
I agree, I want this feature also. I want to configure full discharge of my battery when price is at the highest and not below 250 öre. Since the discharge takes about 1 hour it seems contraproductive to get the 23-cheapest and reverse on/off. Besides, I'd have to add another check to get the "minimum sell price". 

Also, the visualizations in HA would be nicer to have an inverted cheapest hour graph.

**Comment by Ferforfaen** [comment]
I would like that feature also
